src/modules/expenses/expense.routes.ts

```diff
@@ -0,0 +1,84 @@
+import { createRoute } from "@hono/zod-openapi";
+import { z } from "zod";
+
+import { AuthRoles } from "@/common/enums";
+import jsonContentRequired from "@/common/helpers/json-content-required.helper";
+import { jsonContent } from "@/common/helpers/json-content.helper";
+import {
+  authMiddleware,
+  checkRoleGuard,
+  requireAuth,
+} from "@/common/middlewares/auth.middleware";
+import * as HTTPStatusCodes from "@/common/utils/http-status-codes.util";
+import {
+  insertExpenseSchema,
+  selectExpenseSchema,
+} from "@/db/schemas/expense.model";
+
+const tags = ["Expenses"];
+
+export const createExpenseRoute = createRoute({
+  tags,
+  method: "post",
+  path: "/expenses",
+  middleware: [
+    authMiddleware(),
+    requireAuth(),
+    checkRoleGuard(AuthRoles.ADMIN, AuthRoles.USER),
```

Comment:
We do not need to check the role since an expense can be created by any logged in user

Reply:
Right no need. Removed